### PR TITLE
Adjust docstrings

### PR DESCRIPTION
**Pull request by adamsjohanna**
Adjust and add docstrings for all modules used in the advanced orders tutorial (in process).

### CODE REVIEW

File: assume/common/forecasts.py [hunk 1]
```diff
@@ -169,6 +174,9 @@ def calc_forecast_if_needed(self):
 
         This method calculates additional forecasts if they do not already exist, including
         "price_EOM" and "residual_load_forecast".
+
+        Args:
+            self: The instance of CsvForecaster.
```
Comment (maurerle) [inline assume/common/forecasts.py:179]: Self should not be part of the args anywhere - as this is implicit for classes

File: assume/common/forecasts.py [hunk 2]
```diff
@@ -15,8 +15,8 @@ class Forecaster:
     to retrieve forecasts for specific columns, availability of units, and prices of fuel types, returning
     the corresponding timeseries as pandas Series.
 
-    Args:
-    - index (pd.Series): The index of the forecasts.
+    Attributes:
+        index (pd.Series): The index of the forecasts.
```
Comment (maurerle) [inline assume/common/forecasts.py:19]: It is correct to have this called Args:

If you look at this example:
https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

the def __init__ is documented in the class doc string with Args

You would use Attributes for Class-Variables which are independent of the actual instance

File: assume/markets/clearing_algorithms/complex_clearing.py [hunk 3]
```diff
@@ -341,22 +360,24 @@ def clear(
...
+    Calculates the surplus of an order given the market clearing prices and results of the market clearing.
 
     Args:
         order (dict): The order
-        market_clearing_prices (dict): The market clearing prices
-        instance (pyomo.ConcreteModel): The solved pyomo model containing the results of the market clearing
-        children (list[dict]): The linked child bids of the given order
+        market_clearing_prices (dict): The market clearing prices.
+        instance (pyomo.ConcreteModel): The solved pyomo model containing the results of the market clearing.
+        children (list[dict]): The linked child bids of the given order.
```
Comment (maurerle) [inline assume/markets/clearing_algorithms/complex_clearing.py:368]: I do not think punctuation is required here, but I don't mind :smile:

File: assume/common/forecasts.py (hunk 2 above)
Comment (adamsjohanna) [reply assume/common/forecasts.py:19]: So it's not double, if it is added in both class definition and init()? Then I will add it for all others too

Comment (maurerle) [reply assume/common/forecasts.py:19]: yes - Attributes document the class attributes and args document the arguments to the constructor.

In most cases those are quite similar - as args to the constructor are then set as an attribute, so yes you can add them

**Review by maurerle — CHANGES_REQUESTED** [review]
I just pushed a commit which removes the docs for the self args using the regex search and replace of vscodium:

search for:
```
        Args:
            self:.*
```

replace with:
```
        Args:
```
globally :)

but you can look through it

File: assume/common/forecasts.py (hunk 2 above)
Comment (adamsjohanna) [reply assume/common/forecasts.py:19]: Thanks for the explanation 👍

File: assume/markets/clearing_algorithms/complex_clearing.py (hunk 3 above)
Comment (maurerle) [reply assume/markets/clearing_algorithms/complex_clearing.py:368]: I would rather not have any punctuation in the args sections - as it is not a real sentence anyway :smile: 
But thats probably the least important thing

**Review by maurerle — APPROVED** [review]
Looks fine
You could build the docs once to confirm it looks nice if you haven't done that already.

So feel free to merge

**Pull request merged**